generator/main.go

```diff
@@ -0,0 +1,485 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"go.universe.tf/metallb/api/v1beta1"
+	"go.universe.tf/metallb/api/v1beta2"
+	"go.universe.tf/metallb/internal/config"
+	"go.universe.tf/metallb/internal/pointer"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	resourcesNameSpace = "metallb-system"
+	spacer             = "---\n"
+	autoGenComment     = "# This was autogenerated by MetalLB's custom resource generator.\n"
+)
+
+var (
+	inputDirPath = "/var/input"
+)
+
+func main() {
+	var origin = flag.String("origin", "./config.yaml", "name of the configmap file to covnert")
+	flag.Parse()
+
+	resourcesYAML, err := generate(*origin)
+	if err != nil {
+		fmt.Println(err.Error())
```

Comment:
does this mean when an error occurs we redirect it to the file :sweat_smile: ?
if so I rather we create the output file from here, not relying on the user to redirect from outside, maybe even adding some logs along the way - e.g commit id, "reading file.../ generated ...". wdyt?

Reply:
missed that, you're right. I'll create the output file and add logs.

Reply:
you can use log.Fatal that prints the error and exits.